### PR TITLE
fix trainer distributed attributes

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixing typos in exist check...
Fixes #4990

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes [if needed]?
- [x] Did you write any new necessary tests [no need for typos, docs]? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone are aligned!**
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by carmocca** [comment]
Why do we need the `is not None` checks? `==` works for `None`

File: pytorch_lightning/trainer/deprecated_api.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ class DeprecatedDistDeviceAttributes:
     @property
     def on_cpu(self) -> bool:
         # rank_zero_warn("Internal: `on_cpu` is deprecated in v1.1 and will be removed in v1.2.", DeprecationWarning)
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/deprecated_api.py:26]: not related to PR, but should't the deprecation and removal here be 0.2 apart??

**Comment by Borda** [comment]
> Why do we need the `is not None` checks? `==` works for `None`

yes, but the enum does not like being compared with None...
anyway, it is just a deprecation solution :]

File: pytorch_lightning/trainer/deprecated_api.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/trainer/deprecated_api.py:26]: yes, it is in #5297 #5300

**Comment by carmocca** [comment]
> yes, but the enum does not like being compared with None

Should work:

```python
from enum import Enum

class Foo(Enum):
    BAR = "bar"

print(None == Foo.BAR)  # False
``

**Comment by Borda** [comment]
> > yes, but the enum does not like being compared with None
> 
> Should work:
> 
> ```python
> from enum import Enum
> 
> class Foo(Enum):
>     BAR = "bar"
> 
> print(None == Foo.BAR)  # False
> ``
> ```

yes, just tried in the past commit c618ee0

**Review by carmocca — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**